File: Flee/ListClass.vb
Public Class ListClass

	Public type As String
	Public name As String
	Public properties As List(Of ListProperty) = New List(Of ListProperty)

	Public Sub New(header As String, lines As List(Of String))
		Dim header_parts() As String = header.Split(" ")
		Me.type = header_parts(0)
		Me.name = header_parts(1)
		For Each line As String In lines
			If line.Length() = 0 OrElse line(0) = "#" Then
				Continue For
			End If
			If Not line.Contains("=") Then
				Throw New Exception("Invalid property: " & line)
			End If
			properties.Add(New ListProperty(line))
		Next
	End Sub

End Class

File: Flee/ListClass.cs
using System;
using System.Collections.Generic;
using Microsoft.VisualBasic.CompilerServices;

namespace Flee {
	public class ListClass {
		public string type;
		public string name;
		public List<ListProperty> properties = new List<ListProperty>();

		public ListClass(string header, List<string> lines) {
			var header_parts = header.Split(' ');
			type = header_parts[0];
			name = header_parts[1];
			foreach (string line in lines) {
				if (line.Length == 0 || line[0] == '#')
					continue;

				if (!line.Contains("="))
					throw new Exception("Invalid property: " + line);

				properties.Add(new ListProperty(line));
			}
		}
	}
}